devices/src/virtio/mmio.rs

```diff
@@ -338,7 +380,12 @@ impl BusDevice for MmioDevice {
                         if self
                             .check_driver_status(DEVICE_DRIVER, DEVICE_FEATURES_OK | DEVICE_FAILED)
                         {
-                            self.device.ack_features(self.acked_features_select, v);
+                            // First, set the device available features as those acknowledged by driver.
+                            // This should result in ack_features == self.acked_features_select + v.
+                            self.device
```

Comment:
This is incorrect.
`self.acked_features_select` hold the value for selecting either the upper or lower 32bit dword of the `acked_features`.

Don't change `avail_features` at this step of the virtio config.